hie-plugin-api/Haskell/Ide/Engine/Ghc.hs

```diff
@@ -130,131 +127,198 @@ srcErrToDiag df rfm se = do
+
+            -- set the session before we cache the module, so that deferred
+            -- responses triggered by cacheModule can access it
+
+            Session sess <- GhcT pure
+            modifyMTS (\s -> s {ghcSession = Just sess})
+--            cacheModules rfm ts
+            cacheModules rfm [_tm]
+            debugm "setTypecheckedModule: done"
+
+          Nothing -> do
+            debugm $ "setTypecheckedModule: Didn't get typechecked or parsed module for: " ++ show fp
+            --debugm $ "setTypecheckedModule: errs: " ++ show errs
+            failModule fp
+
+        -- Turn any fatal exceptions thrown by GHC into a diagnostic for
+        -- this module so it appears somewhere permanent in the UI.
+        let diags2 =
+              case mtypechecked_module of
+                Nothing ->
+                  let sev = Just DsError
+                      range = Range (Position 0 0) (Position 1 0)
+                      msgTxt = T.unlines errs
+                      d = Diagnostic range sev Nothing (Just "bios") msgTxt Nothing
+                  in Map.insertWith Set.union canonUri (Set.singleton d) diags
+                Just {} -> diags
+
+        return $ IdeResultOk (Diagnostics diags2,errs)
+
+-- TODO: make this work for all components
 cabalModuleGraphs :: IdeGhcM [GM.GmModuleGraph]
-cabalModuleGraphs = doCabalModuleGraphs
-  where
-    doCabalModuleGraphs :: (GM.IOish m) => GM.GhcModT m [GM.GmModuleGraph]
-    doCabalModuleGraphs = do
-      crdl <- GM.cradle
-      case GM.cradleCabalFile crdl of
-        Just _ -> do
-          mcs <- GM.cabalResolvedComponents
-          let graph = map GM.gmcHomeModuleGraph $ Map.elems mcs
-          return graph
-        Nothing -> return []
+cabalModuleGraphs = do
+  mg <- getModuleGraph
+  let (graph, _) = moduleGraphNodes False (Compat.mgModSummaries mg)
+      msToModulePath ms =
+        case ml_hs_file (ms_location ms) of
+          Nothing -> []
+          Just fp -> [ModulePath mn fp]
+        where mn = moduleName (ms_mod ms)
+      nodeMap = IM.fromList [(node_key n,n) | n <- nodes]
+      nodes = verticesG graph
+      gmg = Map.fromList
+              [(mp,Set.fromList deps)
+                  | node <- nodes
+                  , mp <- msToModulePath (node_payload node)
+                  , let int_deps = node_dependencies node
+                        deps = [ d | i <- int_deps
+                                   , Just dep_node <- pure $ IM.lookup i nodeMap
+                                   , d <- msToModulePath (node_payload dep_node)
+                               ]
+                  ]
+  pure [GmModuleGraph gmg]
 
 -- ---------------------------------------------------------------------
 
 makeRevRedirMapFunc :: IdeGhcM (FilePath -> FilePath)
-makeRevRedirMapFunc = make
-  where
-    make :: (GM.IOish m) => GM.GhcModT m (FilePath -> FilePath)
-    make = GM.mkRevRedirMapFunc
+makeRevRedirMapFunc = reverseFileMap
```

Comment:
I think it would be cleaner to remove this function and replace all occurrences by `reversFileMap`

Reply:
This is part of the pluginApi and used by HaRe. Still delete?